src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        if (fromFileName == null) {
+            throw new IllegalArgumentException("File is not found!");
+        }
+
+        Integer supplyCount = 0;
+        Integer buyCount = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName))) {
+            String value;
+            while ((value = br.readLine()) != null) {
+                try {
+                    String [] elements = value.split(",");
+                    if (elements[0].equals("supply")) {
+                        supplyCount += Integer.parseInt(elements[1]);
+                    } else {
+                        buyCount += Integer.parseInt(elements[1]);
+                    }
+                } catch (NumberFormatException n) {
+                    throw new NumberFormatException("Bad format of data!");
+                }
```

Comment:
Instead of rethrowing a new NumberFormatException, consider logging the error or providing more context in the message to help identify the problematic line in the input file.